webui/react/src/components/LogViewerTimestamp.tsx

```diff
@@ -0,0 +1,656 @@
+import { Button, notification, Space, Tooltip } from 'antd';
+import dayjs, { Dayjs } from 'dayjs';
+import React, {
+  useCallback, useEffect, useLayoutEffect, useMemo, useRef, useState,
+} from 'react';
+import screenfull from 'screenfull';
+import { sprintf } from 'sprintf-js';
+import { throttle } from 'throttle-debounce';
+
+import Icon from 'components/Icon';
+import usePrevious from 'hooks/usePrevious';
+import useResize, { DEFAULT_RESIZE_THROTTLE_TIME } from 'hooks/useResize';
+import useScroll, { defaultScrollInfo } from 'hooks/useScroll';
+import { LogLevel, TrialLog } from 'types';
+import { formatDatetime } from 'utils/date';
+import { ansiToHtml, copyToClipboard, toRem } from 'utils/dom';
+import { capitalize } from 'utils/string';
+
+import { FetchArgs } from '../../../api-ts-sdk';
+import { TrialLogFiltersInterface } from '../pages/TrialLogs/TrialLogFilters';
+import { consumeStream } from '../services/utils';
+
+import css from './LogViewer.module.scss';
+import Page, { Props as PageProps } from './Page';
+
+export interface LogViewerTimestampFilterComponentProp {
+  filter: TrialLogFiltersInterface;
+  filterOptions: TrialLogFiltersInterface;
+  onChange?: (newFilters: TrialLogFiltersInterface) => void;
+}
```

Comment:
**suggestion:** This should probably go in `TrialLogFilters.tsx` as:
```
export interface Props {
  filter: TrialLogFiltersInterface;
  filterOptions: TrialLogFiltersInterface;
  onChange?: (newFilters: TrialLogFiltersInterface) => void;
}
```
and then import it here as:
```
import { TrialLogFiltersInterface, Props as TrialLogFiltersProps } from 'pages/TrialLogs/TrialLogFilters';
```

Reply:
Whoops, 200% better.